waffle-chai/src/matchers/ethCalled.ts

```diff
@@ -0,0 +1,56 @@
+import {Contract} from 'ethers';
+import {MockProvider, RecordedCall} from '@ethereum-waffle/provider';
+
+interface AssertionResultInterface {
+  wasCalled: boolean;
+  errorMessage: string;
+  negatedErrorMessage: string;
+}
+
+type IsInCallHistory = (call: RecordedCall) => boolean
+
+export function supportEthCalled(Assertion: Chai.AssertionStatic) {
+  Assertion.addProperty('ethCalled', function () {
+    const arg = this._obj;
+    const [contract, fnName] = Array.isArray(arg) ? arg : [arg];
+
+    if (!(contract instanceof Contract)) {
+      throw new TypeError('ethCalled: argument must be a contract');
+    }
+    const provider = contract.provider;
+    if (!(provider instanceof MockProvider)) {
+      throw new TypeError('ethCalled: contract.provider must be a MockProvider');
+    }
+    if (fnName !== undefined && typeof fnName !== 'string') {
+      throw new TypeError('ethCalled: function name must be a string');
+    }
+
+    const fnSighash = fnName && contract.interface.functions[fnName].sighash;
+
+    const isInCallHistory = (cb: IsInCallHistory): boolean => provider.callHistory.some(cb);
+
+    const getAssertionResult = (): AssertionResultInterface => {
+      if (fnSighash) {
+        return {
+          wasCalled: isInCallHistory(call => call.data === fnSighash),
```

Comment:
this check is flawed. it will work for a function without arguments, but there are two problems.
1) If a function accepts arguments `call.data` contains more stuff, so you need `.startsWith`
2) You also need to check for the address since the function could be called on a different contract